### PR TITLE
Added a second trace for global init checker showing creation of mutable fields

### PR DESCRIPTION
**Pull request by EnzeXing**

### CODE REVIEW

**Review by olhotak — COMMENTED** [review]
I'd suggest to not print the trace if the length of the trace is only 1.

Otherwise this LGTM.

File: tests/init-global/warn/global-irrelevance3.check [hunk 1]
```diff
@@ -9,3 +9,4 @@
   |│           ^^^^^^^
   |└── (() => x) // warn	[ global-irrelevance3.scala:9 ]
   |           ^
+  |The mutable state is created through:
```
Comment (liufengyun) [inline tests/init-global/warn/global-irrelevance3.check:12]: This one is empty --- could be a latent problem with the existing trace.

File: compiler/src/dotty/tools/dotc/transform/init/Objects.scala [hunk 2]
```diff
@@ -1758,19 +1760,23 @@ class Objects:
...
   val readErrorSet: mutable.Set[(ClassSymbol, ClassSymbol)] = mutable.Set.empty
-  def errorReadOtherStaticObject(currentObj: ClassSymbol, otherObj: ClassSymbol)(using Trace, Context) =
+  def errorReadOtherStaticObject(currentObj: ClassSymbol, addr: Heap.Addr)(using Trace, Context) =
+    val otherObj = addr.owner
     if readErrorSet.add((currentObj, otherObj)) then
       val msg =
         "Reading mutable state of " + otherObj.show + " during initialization of " + currentObj.show + ".\n" +
-        "Reading mutable state of other static objects is forbidden as it breaks initialization-time irrelevance. " + Trace.show
+        "Reading mutable state of other static objects is forbidden as it breaks initialization-time irrelevance. " + Trace.show +
+        "The mutable state is created through: " + Trace.show(using addr.getTrace)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/init/Objects.scala:1780]: Minor: Use the following method to change the duplicate colon.

https://github.com/scala/scala3/blob/main/compiler/src/dotty/tools/dotc/transform/init/Trace.scala#L45

**Review by liufengyun — APPROVED** [review]
It's great to have this capability :+1:

**Review by olhotak — APPROVED** [review]
LGTM

**Pull request merged**